partiql-types/src/main/kotlin/org/partiql/types/StaticType.kt

```diff
@@ -80,7 +81,8 @@ sealed class StaticType {
             LIST,
             SEXP,
             STRUCT,
-            BAG
+            BAG,
+            SCHEMA,
```

Comment:
Schema isn't a StaticType. This conflates types with type parameters.

Consider pulling type parameters out of StaticType. That is, a _type_ can be a type parameter (ie `bag<int>`) but a parameter is not necessarily a type (decimal precision/scale, timezone, length constraints on fixed/variable string types)

```kotlin
sealed class StaticType {
  // ... types
}

sealed class TypeParam {

    class Type(type: StaticType) : TypeParam() { ... }

    class Literal(value: Ion) : TypeParam() { ... }

    class Schema( ... ): TypeParam() { ... }
}
```

On the other hand, StaticType is written is a very strange way.. it's an abstract class whose default methods switch on the concrete type to invoke a desired behavior. Rather than just overriding where a concrete type's behavior differents from the abstract type.. We will be replacing this.

Reply:
Referring our offline discussion, this was a model that I was experimenting—while the PR is still in draft–to see how it plays out in the codebase w/different scenarios. I think we're aligned that `StaticType` doesn't follow a more detailed specification and b/c of that there are oddities; in my view, one example is the constraints that we're considering for types and the way we're specifying them. E.g. [`contentClosed` and `primaryKeyFields`](https://github.com/partiql/partiql-lang-kotlin/blob/c4f061b0fb12c872d55ef1b857171dac9af4539a/partiql-types/src/main/kotlin/org/partiql/types/StaticType.kt#L432-L433) are just some parameters on `StructType`which I don't believe in itself is an issue but the problem is it seems adhoc and following no rules.

While we're figuring out our type system specification in more details, to unblock our upcoming project I'm considering a `SchemaType` as just a bag of `StaticType` and defining the missing tuple constraints that we need for now (E.g. `PARTITION_KEY`). This won't define a new type per se like e.g. `BagType()` and is literally bag type itself.